src/clarity-angular/data/datagrid/datagrid-row.ts

```diff
@@ -41,7 +41,7 @@ let nbRow: number = 0;
                          class="datagrid-select datagrid-fixed-column">
                 <div class="radio">
                     <input type="radio" [id]="id" [name]="selection.id + '-radio'" [value]="item"
-                           [(ngModel)]="selection.currentSingle">
+                           [(ngModel)]="selection.currentSingle" [checked]="selection.currentSingle === item">
```

Comment:
Without this, the state of the input doesn't properly resolve between pages. I believe this is due to ngModel forcing a second change detection cycle. https://github.com/angular/angular/blob/5.1.0/packages/forms/src/directives/ng_model.ts#L30